src/pi_cam.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file
+ * @brief     Implementation for Pi camera
+ * @author    Willem de Groot, Arsalan Anwari
+ * @license   See LICENSE
+ */
+#include "pi_cam.hpp"
+
+PiCam::PiCam(const char *filePath, const Vector2D resolution, int fps)
+    : Camera(fps, resolution), filePath(std::string(filePath)), buffer(resolution.getX() * resolution.getY() * 3) {

```

Comment:
`filePath(std::string(filePath))` is inefficient, running `filePath(filePath)` should be enough. That way the constructor `string(const char*)` is called of class member `filePath`.